product_code_unique/product.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from openerp.osv import osv
+from openerp.tools.translate import _
+
+
+class product_code_unique_product(osv.osv):
+    _name = "product.product"
+    _inherit = "product.product"
+
+    def _check_defaultcode_and_variants(self, cr, uid, ids):
+        for product in self.browse(cr, uid, ids):
```

Comment:
Actually browse was proved to be more efficient in most cases (because of the prefetching if I remmeber correctly)